File: server-py/metrics.py
import time

_start_time = time.time()
_request_count = 0
_total_response_ms = 0.0


def record_request(ms: float) -> None:
    global _request_count, _total_response_ms
    _request_count += 1
    _total_response_ms += ms


def get_metrics() -> dict:
    return {
        "uptimeSeconds": int(time.time() - _start_time),
        "requestCount": _request_count,
        "avgResponseMs": round(_total_response_ms / _request_count) if _request_count else 0,
    }

File: server/metrics.ts
const startTime = Date.now()
let requestCount = 0
let totalResponseMs = 0

export function recordRequest(ms: number) {
  requestCount++
  totalResponseMs += ms
}

export function getMetrics() {
  return {
    uptimeSeconds: Math.floor((Date.now() - startTime) / 1000),
    requestCount,
    avgResponseMs: requestCount > 0 ? Math.round(totalResponseMs / requestCount) : 0,
  }
}
